cli/ballerina-packerina/src/test/java/org/ballerinalang/packerina/cmd/DependencyScopeTest.java

```diff
@@ -0,0 +1,141 @@
+                resolve(ProjectDirConstants.TARGET_BALO_DIRECTORY);
+        File baloFile = new File(balo.toString() + File.separator + baloFileName);
+        File baloZipFile = new File(balo.toString() + File.separator +
+                baloFileName.concat(".zip"));
+        Path scopeToml = Paths.get(testResourcesURI).resolve("scope-toml");
+        String storedJarPath = "platform-libs/storedJar.jar";
+
+        // Build the project
+        String[] compileArgs = {"--all", "--skip-tests"};
+        BuildCommand buildCommand = new BuildCommand(this.testResources, printStream, printStream,
+                false, true);
+        new CommandLine(buildCommand).parse(compileArgs);
+        // default scope
+        buildCommand.execute();
+        Assert.assertTrue(Files.exists(balo), "Check if balo directory exists");
+
+        // Check whether dependency jars getting packed to balo
+        Assert.assertTrue(renameFile(baloFile, baloZipFile));
+        Assert.assertTrue(isJarExists(baloZipFile, storedJarPath));
+
+        // package scope
+        copy(scopeToml.resolve("case1").resolve(ProjectDirConstants.MANIFEST_FILE_NAME),
+                this.testResources.resolve(ProjectDirConstants.MANIFEST_FILE_NAME));
+        buildCommand.execute();
+        Assert.assertTrue(Files.exists(balo), "Check if balo directory exists");
+
+        // Check whether dependency jars getting packed to balo
+        Assert.assertTrue(renameFile(baloFile, baloZipFile));
+        Assert.assertTrue(isJarExists(baloZipFile, storedJarPath));
+
+        // package scope
+        copy(scopeToml.resolve("case2").resolve(ProjectDirConstants.MANIFEST_FILE_NAME),
+                this.testResources.resolve(ProjectDirConstants.MANIFEST_FILE_NAME));
+        buildCommand.execute();
+        Assert.assertTrue(Files.exists(balo), "Check if balo directory exists");
+
+        // Check whether dependency jars getting packed to balo
+        Assert.assertTrue(renameFile(baloFile, baloZipFile));
+        Assert.assertFalse(isJarExists(baloZipFile, storedJarPath));
+
+        // compile scope
+        copy(scopeToml.resolve("case3").resolve(ProjectDirConstants.MANIFEST_FILE_NAME),
+                this.testResources.resolve(ProjectDirConstants.MANIFEST_FILE_NAME));
+        buildCommand.execute();
+        Assert.assertTrue(Files.exists(balo), "Check if balo directory exists");
+
+        // Check whether dependency jars getting packed to balo
+        Assert.assertTrue(renameFile(baloFile, baloZipFile));
+        Assert.assertFalse(isJarExists(baloZipFile, storedJarPath));
+    }
+
+    private boolean renameFile(File oldName, File newName) {
+        return oldName.renameTo(newName);
+    }
+
+    private boolean isJarExists (File file, String jarEntry) throws IOException {
+        try (ZipFile zipFile = new ZipFile(file)) {
+            Enumeration<? extends ZipEntry> entries = zipFile.entries();
```

Comment:
No need to iterate all the zip entries and find whether a particular entry is there or not. The time complexity and performance hit would be high by doing this way. Instead, you can directly check a particular entry's presence in the following way.

`return new ZipFile(file).getEntry(jarEntry) != null;`